### PR TITLE
sql: deflake TestInternalExecAppNameInitialization

### LINKED ISSUE #33499: sql: TestInternalExecAppNameInitialization failed under stress
SHA: https://github.com/cockroachdb/cockroach/commits/d6adf24cae788d7cd967feadae8e9c0388ce5273

Parameters:
```
TAGS=
GOFLAGS=-race
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestInternalExecAppNameInitialization PKG=github.com/cockroachdb/cockroach/pkg/sql TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1080705&tab=buildLog

```
  github.com/cockroachdb/cockroach/pkg/sql.(*LeaseManager).DeleteOrphanedLeases.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/lease.go:1791 +0x392
  github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:198 +0x14a

Goroutine 527 (finished) created at:
  github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc.(*Server).serveStreams.func1()
      /go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:678 +0xb8
  github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/transport.(*http2Server).operateHeaders()
      /go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/transport/http2_server.go:424 +0x1483
  github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/transport.(*http2Server).HandleStreams()
      /go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/transport/http2_server.go:464 +0x87a
  github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc.(*Server).serveStreams()
      /go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:676 +0x16e
  github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc.(*Server).handleRawConn.func2()
      /go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:631 +0x53
  github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc.(*Server).handleRawConn.func3()
      /go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:639 +0x34
==================
I190104 08:01:50.851719 62766 sql/event_log.go:131  [n1] Event: "node_join", target: 1, info: {Descriptor:{NodeID:1 Address:127.54.16.172:34185 Attrs: Locality: ServerVersion:2.1-3 BuildTag:v2.2.0-alpha.20181217-398-gd6adf24 StartedAt:1546588909761569692 LocalityAddress:[] XXX_NoUnkeyedLiteral:{} XXX_sizecache:0} ClusterID:a8cd05af-99cd-4d58-960e-2a4ac1035f15 StartedAt:1546588909761569692 LastUp:1546588909761569692}
I190104 08:01:50.890771 62546 util/stop/stopper.go:536  quiescing; tasks left:
2      node.Node: batch
1      kv.DistSender: sending partial batch
1      [async] transport racer
1      [async] storage.split: processing replica
1      [async] kv.TxnCoordSender: heartbeat loop
1      [async] closedts-rangefeed-subscriber
I190104 08:01:50.891393 62546 util/stop/stopper.go:536  quiescing; tasks left:
1      kv.DistSender: sending partial batch
1      [async] transport racer
1      [async] storage.split: processing replica
1      [async] kv.TxnCoordSender: heartbeat loop
1      [async] closedts-rangefeed-subscriber
I190104 08:01:50.891874 62546 util/stop/stopper.go:536  quiescing; tasks left:
1      kv.DistSender: sending partial batch
1      [async] transport racer
1      [async] storage.split: processing replica
1      [async] kv.TxnCoordSender: heartbeat loop
I190104 08:01:50.892331 62546 util/stop/stopper.go:536  quiescing; tasks left:
1      kv.DistSender: sending partial batch
1      [async] transport racer
1      [async] storage.split: processing replica
W190104 08:01:50.892567 62417 internal/client/txn.go:507  [n1,split,s1,r19/1:/{Table/23-Max}] failure aborting transaction: node unavailable; try another peer; abort caused by: result is ambiguous (server shutdown)
I190104 08:01:50.892842 62546 util/stop/stopper.go:536  quiescing; tasks left:
1      [async] transport racer
1      [async] storage.split: processing replica
E190104 08:01:50.893222 62417 storage/queue.go:844  [n1,split,s1,r19/1:/{Table/23-Max}] unable to split [n1,s1,r19/1:/{Table/23-Max}] at key "/Table/24": split at key /Table/24 failed: result is ambiguous (server shutdown)
I190104 08:01:50.893650 62546 util/stop/stopper.go:536  quiescing; tasks left:
1      [async] transport racer
I190104 08:01:50.909915 62508 kv/transport_race.go:113  transport race promotion: ran 19 iterations on up to 261 requests
    --- FAIL: TestInternalExecAppNameInitialization/session_bound_exec (1.31s)
    	testing.go:730: race detected during execution of test
	testing.go:730: race detected during execution of test

```

### PR DESCRIPTION
**Pull request by petermattis**
Use the fully qualified name `system.public.lease` for the internal queries performed by the table lease manager. This avoids the work to qualify `system.lease` if one of these internal queries is run concurrently with `SHOW QUERIES`. Qualifying table names by `SHOW QUERIES` results in an "innocuous race" with execution of the query (the race should be separately fixed, but is outside of the scope of this change).

Fixes #33499

Release note: None

### CODE REVIEW

**Comment by petermattis** [comment]
I understand why this change avoids the race, but not why #33506 does as well. AFAICT, the race is between the orphaned table leases query execution and `SHOW QUERIES` finding that query and formatting it. Why would switching to manual replication avoid the orphaned table leases query? I'd like to figure that out before merging this.

File: pkg/sql/internal_test.go [hunk 1]
```diff
@@ -150,6 +150,11 @@ func TestInternalExecAppNameInitialization(t *testing.T) {
 
 	params, _ := tests.CreateTestServerParams()
 	params.Insecure = true
+	params.Knobs.SQLLeaseManager = &sql.LeaseManagerTestingKnobs{
+		// Disable the orphaned leases query which has an innocuous race with the
+		// SHOW QUERIES query used elsewhere in this test.
+		DisableDeleteOrphanedLeases: true,
```
Comment (knz) [inline pkg/sql/internal_test.go]: you probably want to do the same in `run_control_test.go` since the tests in there are running into the same issue

Comment (petermattis) [reply pkg/sql/internal_test.go]: Will do once I fully understand what is going on.

**Comment by petermattis** [comment]
Rather than disabling orphaned table lease deletion, I've changed table lease manager to use the fully qualified `system.public.lease` table name. This does not eliminate the general race between query execution and `SHOW QUERIES` formatting of running queries, but it does avoid it for the usage by table lease manager.

@andreimatei, @knz, @vivekmenezes PTAL

**Review by knz — COMMENTED** [review]
works for me.
I suspect the tests in `run_control_test.go` can be simplified to not require disabling replication now.

**Comment by petermattis** [comment]
> I suspect the tests in run_control_test.go can be simplified to not require disabling replication now.

I took a quick look, and at least one `TestCancelDistSQLQuery` needs manual replication.

**Review by vivekmenezes — COMMENTED** [review]
the changes to lease.go LGTM

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/33562#-:-LVicNvE4jBGAnQeH9zu:b9j60w4)* status: :shipit: complete! 0 of 0 LGTMs obtained

**Review by andreimatei — COMMENTED** [review]
Just so I'm clear - for most of the tests in question manual replication was never needed, was it? What was "needed" was a `TestCluster` instead of a `TestServer`, right?

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/33562#-:-LVid5eu-iU7KKlOT78j:bkb5shv)* status: :shipit: complete! 0 of 0 LGTMs obtained

**Review by petermattis — COMMENTED** [review]
> Just so I'm clear - for most of the tests in question manual replication was never needed, was it? What was "needed" was a TestCluster instead of a TestServer, right?

Yes.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/33562#-:-LVie2_q3Po6fxX01xVw:b-dchqd6)* status: :shipit: complete! 0 of 0 LGTMs obtained

**Comment by petermattis** [comment]
I'm bors'ing because this test is flaky. I'll follow-up with another PR to use auto replication when possible in `run_control_test.go` (that might not be until tomorrow).

bors r=vivekmenezes

**Pull request merged**